packages/federation/test/gateway.test.ts

```diff
@@ -956,4 +957,133 @@ describe('Federation', () => {
+    //   expect((result as RouterConfig).supergraphSdl).toBeDefined();
+    // });
+
+    it('should pass the variables correctly to the fetch function', async () => {
+      await getStitchedSchemaFromManagedFederation({
+        apiKey: 'test-api-key',
+        graphId: 'test-graph-id',
+        lastSeenId: 'test-last-seen-id',
+        upLink: 'test-up-link',
+        fetch(url, bodyInit) {
+          expect(url).toBe('test-up-link');
+          expect(bodyInit?.body).toContain('"lastSeenId":"test-last-seen-id"');
+          expect(bodyInit?.body).toContain('"graphId":"test-graph-id"');
+          expect(bodyInit?.body).toContain('"apiKey":"test-api-key"');
+          return Promise.resolve(
+            new Response(
+              JSON.stringify({
+                data: {
+                  routerConfig: {
+                    __typename: 'RouterConfigResult',
+                    minDelaySeconds: 10,
+                    id: 'test-id-1',
+                    supergraphSDL: 'test supergraph sdl',
+                  },
+                },
+              }),
+            ),
+          );
+        },
+      });
+
+      expect.assertions(4);
+    });
+
+    it('should handle unchanged supergraph SDL', async () => {
+      const mochUnchangedSuppergraph = () =>
+        Promise.resolve(
+          new Response(
+            JSON.stringify({
+              data: {
+                routerConfig: {
+                  __typename: 'Unchanged',
+                  minDelaySeconds: 10,
+                  id: 'test-id-1',
+                },
+              },
+            }),
+          ),
+        );
+
+      const result = await getStitchedSchemaFromManagedFederation({
+        apiKey: 'service:fake-key',
+        graphId: 'Spotify-Demo-Graph-nsqe3n',
+        fetch: mochUnchangedSuppergraph,
+      });
+
+      expect(result).toMatchObject({
+        id: 'test-id-1',
+        minDelaySeconds: 10,
+      });
+    });
+
+    it('should handle fetch errors returned by GraphOS', async () => {
+      const mochUnchangedSuppergraph = () =>
+        Promise.resolve(
+          new Response(
+            JSON.stringify({
+              data: {
+                routerConfig: {
+                  __typename: 'FetchError',
+                  minDelaySeconds: 10,
+                  code: 'FETCH_ERROR',
+                  message: 'Test error message',
+                },
+              },
+            }),
+          ),
+        );
+
+      const result = await getStitchedSchemaFromManagedFederation({
+        apiKey: 'service:fake-key',
+        graphId: 'Spotify-Demo-Graph-nsqe3n',
+        fetch: mochUnchangedSuppergraph,
+      });
+
+      expect(result).toMatchObject({
+        minDelaySeconds: 10,
+        error: { code: 'FETCH_ERROR', message: 'Test error message' },
+      });
+    });
+
+    it('should return the supergraph SDL with metadata', async () => {
+      const mochUnchangedSuppergraph = () =>
+        Promise.resolve(
+          new Response(
```

Comment:
```suggestion
          new Response(
```
You can use `Response` from `@whatwg-node/fetch` to make it pass on Node 16. And you can also use `Response.json` instead of `new Response(JSON.stringify(`